Add  /@@test-rendering views to docs

Manas-Kenge opened the pull request:
## Issue number

- Fixes #1871

## Description

This is a WIP, need an initial review to keep working. As per https://github.com/plone/documentation/issues/1871#issuecomment-2717865542 a mention of both test-rendering and Storyboard docs in the migration docs under [Migrating from Plone Classic UI to Volto](https://6.docs.plone.org/backend/upgrading/version-specific-migration/migrate-to-volto.html) is remaining.




<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1903.org.readthedocs.build/classic-ui/theming/create-add-on.html#styles-test-rendering

<!-- readthedocs-preview plone6 end -->

Manas-Kenge committed:
mention of the /@@test-rendering views

Manas-Kenge committed:
Merge branch '6.0' into test-rendering-views

stevepiercy commented on docs/classic-ui/views.md:
````diff
@@ -252,6 +252,60 @@ Add the name of the new view `my-view` to the following list:
+Append one of these view names to any Plone site URL:
+
+```
+http://localhost:8080/Plone/@@test-rendering
````
No. The URL segment to append to the site URL is `/@@test-rendering`. If I append the provided string, I get a 404.

Also convert this to a bulleted list, not a code block.

acsr replied:
@stevepiercy Yep you are right. I was careless during writing copy pasting the underscore based terms too often. I corrected the terms in all ticket comments to avoid further confusion (And I had to do it twice)

acsr replied:
@Manas-Kenge To make it clear: The part to **append** is e.g. just `/@@test-rendering` etc. and leave out the absolute url prefix.

stevepiercy reviewed (changes requested):
I'll do a second review, but for now...

- Please follow https://6.docs.plone.org/contributing/documentation/authors.html#plone-documentation-styleguide, specifically:

  > Headings should be "Sentence cased", not "Title Cased".

- Don't inject content between a target `(classic-ui-template-slots-label)=` and its heading. Keep them close together.
- The correct proper name is "Classic UI" with a space.
- Use 4 spaces indentation in all list items, else they don't render correctly.
- This content breaks flow of the original document. You must consider where it belongs, and not throw it in wherever.
- A picture is worth a thousand words. Screenshots would be nice. See guidance in https://6.docs.plone.org/contributing/documentation/myst-reference.html#images-and-figures

stevepiercy commented:
@plone/classicui-team could I get a review from y'all on content accuracy, and where this documentation should reside? It's in the wrong place at the moment. Thank you!

acsr committed:
Merge branch '6.0' into test-rendering-views

petschki commented on docs/classic-ui/views.md:
```diff
@@ -252,6 +252,60 @@ Add the name of the new view `my-view` to the following list:
+    - Grid layout examples
+    - Form control variants
+    - Navigation components
+    - Interactive element states
```
I'd also mention the colormode switcher here

acsr replied:
@petschki I cannot find any colormode text in the templates. Any hints?

acsr replied:
The page https://6.docs.plone.org/classic-ui/theming/color-mode.html is listed by google, but not present (404).

The Light/dark/auto Toggle Button is mentioned in https://6.docs.plone.org/classic-ui/theming/color-modes.html#toggle-button

I guess you mean "mentioning" means to point out, that the appearance of the examples differs depending on your day/night dark/light theme setting.

This is a halfbaked admonition. @stevepiercy can wordsmith a better one.

```` myst
```{important}
Please take into consideration how to make sure checking the examples includes both the dark and light setting when testing the UI.
```
````

petschki replied:
I just wanted to mention, that you can see the color mode switcher in action here ... if you want to use it for your project you have to read the doc ...

stevepiercy replied:
It's a button interface in the lower right corner.

petschki reviewed (changes requested):
I'd suggest to put this somewhere into `classic-ui/theming/create-add-on.md` as subsection **Styles Test Rendering**

acsr commented on docs/classic-ui/views.md:
````diff
@@ -252,6 +252,60 @@ Add the name of the new view `my-view` to the following list:
+1.  Component Alerts Demo (`@@test-rendering`)
+    Displays Bootstrap-style status message examples:  
+    ```
+    http://localhost:8080/Plone/@@test-rendering
````
same as line 265 here: single line or bullet without code fence, just part to append as inline code, no url prefix.

acsr commented on docs/classic-ui/views.md:
````diff
@@ -252,6 +252,60 @@ Add the name of the new view `my-view` to the following list:
+2. Bootstrap Pattern Library (`@@test-rendering-cheatsheet`)
+   Renders a comprehensive component reference based on Bootstrap's official cheatsheet:
+    ```
+    http://localhost:8080/Plone/@@test-rendering-cheatsheet
````
same as line 265 here: single line or bullet without code fence, just part to append as inline code, no url prefix.

acsr commented on docs/classic-ui/views.md:
````diff
@@ -252,6 +252,60 @@ Add the name of the new view `my-view` to the following list:
+3. Icon Showcase (`@@test-rendering-icons`)
+   Presents all icon classes available in the current theme:
+    ```
+    http://localhost:8080/Plone/@@test-rendering-icons
````
same as line 265 here: single line or bullet without code fence, just part to append as inline code, no url prefix.

acsr reviewed (changes requested):
@Manas-Kenge Thx for this initial iteration. Maybe we need another round.

Manas-Kenge committed:
change location, add screenshots, remove url

Manas-Kenge committed:
Merge branch '6.0' into test-rendering-views

Manas-Kenge committed:
Merge branch '6.0' into test-rendering-views

stevepiercy committed:
Merge branch '6.0' into test-rendering-views

stevepiercy committed:
Merge branch '6.0' into test-rendering-views

jensens commented:
It is indeed important information for theming Classic UI.

acsr commented:
I am lost how to review latest changes due to confusion if a test build exists or how to checkout the latest state of the pr locally. If I am the origin of that mess, get me out of the ditch… ;-)

stevepiercy commented:
@acsr I'm not sure exactly what you want to review, but to see the rendered HTML, visit the pull request preview at https://plone6--1903.org.readthedocs.build/classic-ui/theming/create-add-on.html#styles-test-rendering. If a repo supports pull request previews, then Read the Docs will insert a link in the PR description, which I then usually modify to point to the affected files for convenience.

To checkout this branch, you first need to add a git remote `https://github.com/Manas-Kenge/documentation.git`, then `git fetch --all`, and finally checkout the branch. You can then view git commit history.

acsr commented:
> visit the pull request preview at https://plone6--1903.org.readthedocs.build/classic-ui/theming/create-add-on.html#styles-test-rendering.

I did exactly this (the link at the top of the PR) but it did not work for me. (I had this several times, that these links seem broken. I thought it was due to they were outdated because of changes after a review.)

But it is possible that Github ditches them after some time.

acsr commented on docs/classic-ui/theming/create-add-on.md:
```diff
@@ -141,3 +141,77 @@ $state-colors: map-merge($state-colors, $custom-colors);
+    - Form control variants
+    - Navigation components
+    - Interactive element states
+    - Colormode switcher
```
As I wrote before in my last review: 
**There is no `Colormode switcher` mentioned in the current Plone6.1 cheatsheet at all.** 
Not even the word `color` is present to search on the page. Also no manual dark/light colormode switcher is visible by default in Classic UI, in the personal prefs or as action that could be activated. This needs more explanation or to be removed. 

The `Colormode switcher` item was requested by @petschki very vague [in this review comment](https://github.com/plone/documentation/pull/1903#discussion_r1993683944):
> I'd also mention the colormode switcher here

@petschki  -> Can you please clarify / refine that?

The Light/dark/auto Toggle Button is mentioned in https://6.docs.plone.org/classic-ui/theming/color-modes.html#toggle-button

see also from my comment: https://github.com/plone/documentation/pull/1903#discussion_r1993790568 :

> I guess you mean "mentioning" means to point out, that the appearance of the examples differs depending on your day/night dark/light theme setting.
> 
> This is a halfbaked admonition. @stevepiercy can wordsmith a better one.
> 
> ```{important}
> Please take into consideration how to make sure checking the examples includes both the dark and light setting when testing the UI.
> ```

Manas-Kenge replied:
@acsr I will remove the `Colormode switcher` point, as for Light/dark/auto Toggle there is no need of a mention right? Or should we wait for @petschki to respond?

acsr replied:
@Manas-Kenge Wait one day and then take it out. I push him ;-)

acsr replied:
It can be added in a next iteration again. I have no clue what to add except my suggested admonition.

petschki replied:
The colormode switcher is on  the second tab in the "bootstrap cheatsheet" at the bottom right ... https://classic.demo.plone.org/@@test-rendering-cheatsheet

the implementation is already documented here https://6.docs.plone.org/classic-ui/theming/color-modes.html

stevepiercy replied:
The color mode switcher interface is located in the lower right corner of the browser window.

acsr reviewed (changes requested):
@Manas-Kenge THX! Everything else looks good to me… except the orphaned `Colormode switcher` hint, that needs more love. I suggest taking it out and continue in a seperate iteration issued by @petschki .

stevepiercy commented:
> > visit the pull request preview at https://plone6--1903.org.readthedocs.build/classic-ui/theming/create-add-on.html#styles-test-rendering.
> 
> I did exactly this (the link at the top of the PR) but it did not work for me. (I had this several times, that these links seem broken. I thought it was due to they were outdated because of changes after a review.)
> 
> But it is possible that Github ditches them after some time.

RTD inserts the link immediately when it starts the build, not when the build completes. So if you visit the link within 5 minutes of the PR creation, then you might get a 404. Otherwise, it should always work. It works for me now.

stevepiercy committed:
Merge branch '6.0' into test-rendering-views

stevepiercy committed:
Sentence case headings

stevepiercy committed:
Simplify the structure.
- Avoid enumerating headings.
- Use complete sentences and headings that align with the content.

stevepiercy commented:
I'm happy with this content after making a few commits. @acsr or @petschki one more pass, please? Thank you!

petschki reviewed (dismissed):
good!

acsr reviewed (dismissed):
LGTM @petschki @Manas-Kenge @stevepiercy finally fine…

acsr commented:
Background: I was waiting for over 10 years to have this finally in the docs. There is nothing done unless you do it. (Shame on me). In this case I at least wrote the ticket now. And @Manas-Kenge picked it up. 

When Ramon and Albert were working on Barceloneta they were surprised when I showed it to them in Sorrento. But then I left it to others to make it into the docs.

Manas-Kenge committed:
Merge branch '6.0' into test-rendering-views

stevepiercy reviewed (changes requested):
I just noticed that these images are too wide to display clearly in the portal. Please see https://6.docs.plone.org/contributing/documentation/myst-reference.html#width-of-media.

You can either crop out the left toolbar or collapse it.

I also found a bug and opened an [issue in plone-sphinx-theme](https://github.com/plone/plone-sphinx-theme/issues/71) for it, but that's beyond the scope of this PR.

Manas-Kenge committed:
Merge branch '6.0' into test-rendering-views

Manas-Kenge committed:
crop images

stevepiercy reviewed (changes requested):
Please resize the browser such that images are no wider than 760 pixels, as described in https://6.docs.plone.org/contributing/documentation/myst-reference.html#width-of-media

jensens commented:
There is also for any type with image fields the `@@images-test` view from [plone.namedfile](https://github.com/plone/plone.namedfile).

stevepiercy commented:
> There is also for any type with image fields the `@@images-test` view from [plone.namedfile](https://github.com/plone/plone.namedfile).

https://classic.demo.plone.org/@@images-test 😢 

@jensens got a valid link to view?

jensens commented:
in theory this should work, but its broken, hmm, smells like a bug
https://classic.demo.plone.org/en/demo/an-image.jpg/@@images-test

stevepiercy commented:
> in theory this should work, but its broken, hmm, smells like a bug https://classic.demo.plone.org/en/demo/an-image.jpg/@@images-test

And an old one at that https://github.com/plone/plone.namedfile/issues/125.

Until it's resolved, we shouldn't document it.

stevepiercy commented:
@mauritsvanrees corrected me, and created a new issue https://github.com/plone/plone.namedfile/issues/173.

Manas-Kenge committed:
resize images

stevepiercy reviewed (changes requested):
You need to resize the browser window, then take the screenshot to fit the browser area that matters. Downsizing an image worsens its legibility. Was that not clear in the documentation?

https://6.docs.plone.org/contributing/documentation/myst-reference.html#width-of-media

Manas-Kenge committed:
adjust width

Manas-Kenge commented:
Hey @stevepiercy, I did as you asked, but the image quality hasn't improved even after taking the screenshots.

stevepiercy commented:
You did not resize your browser width. In this screenshot, you can see that I resized my browser width such that the area I want to capture has a width of 760 pixels.

![Screenshot 2025-03-21 at 12 14 14 AM](https://github.com/user-attachments/assets/739d783b-abb9-418d-a78f-d1f8df9dc02c)

I would then get a screenshot of only that area I needed, without all the navigation.

![Screenshot 2025-03-21 at 12 15 33 AM](https://github.com/user-attachments/assets/21585b9e-b08c-4f9b-8957-37420c807db6)

Manas-Kenge committed:
final images

stevepiercy reviewed (approved):
There we go! Thank you! Merging when green.

mauritsvanrees commented:
> in theory this should work, but its broken, hmm, smells like a bug https://classic.demo.plone.org/en/demo/an-image.jpg/@@images-test

I have released Plone 6.1.1 final, and Philip has updated the Classic demo site and this now works.
